source/common/http/http2/metadata_encoder.cc

```diff
@@ -0,0 +1,83 @@
+#include "common/http/http2/metadata_encoder.h"
+
+#include "nghttp2/nghttp2.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http2 {
+
+bool MetadataEncoder::createPayload(MetadataMap& metadata_map) {
+  if (payload_.length() > 0 || metadata_map.empty()) {
+    // |payload_| is not empty or |metadata_map| is empty. Return failure.
+    ENVOY_LOG(error, "payload_ size is {}, metadata_map is {}empty", payload_.length(),
```

Comment:
{}empty -> {} empty

Reply:
Also Envoy style does require error handling be tested.  You may want to change some of your error handling to ASSERT if it should never happen outside of OOM (memory reservation) or make functions virtual so you can mock out returns if you think it's an actual likely failure mode you can't force.

https://99639-65214191-gh.circle-artifacts.com/0/build/envoy/generated/coverage/coverage.source_common_http_http2_metadata_encoder.cc.html

Reply:
Thanks for the review!!

I use "{}empty" so there is only one space between "is" and "empty" when metadata_map.empty() is true, :)

100% coverage is a very good call!! I found a bug: forgot to destroy the deflator in one of the error cases!! Fixed.

Reply:
ping on this. Can we switch this to an ASSERT? Should this ever happen with sane calling code?